### PR TITLE
Add readonly states to all form elements

### PR DESCRIPTION
**Pull request by ynotdraw**
## 🚀 Description

This PR adds `@isReadOnly` support to all controls and fields that support it.

---

## 🔬 How to Test

Visit the following pages and verify the read only states:

- https://202bc927.ember-toucan-core.pages.dev/docs/components/checkbox-field#checkboxfield-with-ischecked-and-isreadonly
- https://202bc927.ember-toucan-core.pages.dev/docs/components/file-input-field#fileinputfield-with-isreadonly
- https://202bc927.ember-toucan-core.pages.dev/docs/components/input-field#inputfield-with-isreadonly-and-a-value
- https://202bc927.ember-toucan-core.pages.dev/docs/components/radio-group-field#radiogroupfield-with-label-and-all-read-only-with-one-checked
- https://202bc927.ember-toucan-core.pages.dev/docs/components/textarea-field#textareafield-with-isreadonly-and-a-value

---

## 📸 Images/Videos of Functionality

I got this visually design-approved with Mary and David 🎉 

Screenshot 2023-04-20 at 11 15 12 AM

Screenshot 2023-04-20 at 11 16 02 AM

Screenshot 2023-04-20 at 11 17 00 AM

Screenshot 2023-04-20 at 11 17 17 AM

Screenshot 2023-04-20 at 11 17 57 AM

Screenshot 2023-04-20 at 11 18 16 AM

### CODE REVIEW

File: test-app/tests/integration/components/checkbox-test.gts [hunk 1]
```diff
@@ -95,6 +107,43 @@ module('Integration | Component | Checkbox', function (hooks) {
...
+    </template>);
+
+    assert.dom('[data-checkbox]').hasClass('bg-titles-and-attributes');
+    assert.dom('[data-checkbox]').hasClass('border-none');
+    assert.dom('[data-checkbox]').hasNoClass('bg-primary-idle');
+    assert.dom('[data-checkbox]').hasNoClass('bg-transparent');
+    assert.dom('[data-checkbox]').hasNoClass('bg-surface-xl');
+    assert.dom('[data-checkbox]').hasNoClass('bg-normal-idle');
+    assert.dom('[data-checkbox]').hasNoClass('bg-disabled');
+    assert.dom('[data-checkbox]').hasNoClass('border-disabled');
```
Comment (ynotdraw) [inline test-app/tests/integration/components/checkbox-test.gts:124]: Visual regression testing when 😅 

This is okay for now so we can ensure we are applying the correct classes, but we wouldn't need these assertions at all with something like Percy.  Maybe soon™.

File: test-app/tests/integration/components/input-field-test.gts [hunk 2]
```diff
@@ -196,4 +196,20 @@ module('Integration | Component | Fields | Input', function (hooks) {
...
     assert.dom('[data-root-field="selector"]').exists();
   });
+
+  test('it disables the input using `@isDisabled`', async function (assert) {
+    await render(<template>
+      <InputField @label="Label" @isDisabled={{true}} data-input />
+    </template>);
+
+    assert.dom('[data-input]').isDisabled();
+  });
```
Comment (ynotdraw) [inline test-app/tests/integration/components/input-field-test.gts:206]: It looked like we didn't have a disabled test like we do with the other components so I added one

File: test-app/tests/integration/components/checkbox-test.gts (hunk 1 above)
Comment (simonihmig) [reply test-app/tests/integration/components/checkbox-test.gts:124]: We are not blocked on introducing Percy on OSS Github repos, are we? They have a free plan, which might be enough, but I think there is also a special program for OSS: https://www.browserstack.com/open-source

**Review by simonihmig — APPROVED** [review]
Nice! 🚀

File: test-app/tests/integration/components/checkbox-test.gts (hunk 1 above)
Comment (ynotdraw) [reply test-app/tests/integration/components/checkbox-test.gts:124]: I don't think so, no! 🎉 We may need to seek approval from someone to add it but I'd imagine it'd be pretty quick. Let me see if I can get that process started today.

File: docs/components/radio-group-field/index.md [hunk 3]
```diff
@@ -263,3 +279,23 @@ Consumers have direct access to the underlying [radio element](https://developer
...
+    <group.RadioField @label='Option 2' @value='option-2' @isReadOnly={{true}} />
+    <group.RadioField @label='Option 3' @value='option-3' @isReadOnly={{true}} />
+  </Form::Fields::RadioGroup>
+</div>
+
+### RadioGroupField with label and all read only with one checked
+
+<div class='mb-4 w-64'>
+  <Form::Fields::RadioGroup @label='Label' @name='read-only-checked' @value="option-2" as |group|>
+    <group.RadioField @label='Option 1' @value='option-1' @isReadOnly={{true}} />
```
Comment (nicolechung) [inline docs/components/radio-group-field/index.md:297]: Question: so for groups, you have to apply readonly to each option, it's not available on `Form::Fields::RadioGroup` as a group?

File: docs/components/checkbox-field/index.md [hunk 4]
```diff
@@ -308,3 +312,22 @@ Target the error block via `data-error`.
     @isDisabled={{true}}
   />
 </div>
+
+### CheckboxField with isReadOnly
```
Comment (nicolechung) [inline docs/components/checkbox-field/index.md:316]: I see the `index.md` is updated here but not for checkbox-group-field?

File: test-app/tests/integration/components/checkbox-test.gts [hunk 5]
```diff
@@ -95,6 +107,43 @@ module('Integration | Component | Checkbox', function (hooks) {
...
   });
 
+  test('it applies the expected classes when `@isChecked={{true}}` and `@isReadOnly={{true}}', async function (assert) {
+    await render(<template>
+      {{! we do not require a label, but instead suggest using Field / TextareaField }}
+      {{! template-lint-disable require-input-label }}
+      <CheckboxControl @isChecked={{true}} @isReadOnly={{true}} data-checkbox />
+    </template>);
+
+    assert.dom('[data-checkbox]').hasClass('bg-titles-and-attributes');
```
Comment (nicolechung) [inline test-app/tests/integration/components/checkbox-test.gts:117]: Very thorough! It's too bad for cases like this you can't pass an array of classes in to check for...

**Review by nicolechung — APPROVED** [review]

File: docs/components/radio-group-field/index.md (hunk 3 above)
Comment (ynotdraw) [reply docs/components/radio-group-field/index.md:317]: Excellent suggestion - added it to the group at https://github.com/CrowdStrike/ember-toucan-core/pull/132/commits/5f6f25084e54f15d1a51da1f1cc525ad3c6c212a

File: docs/components/checkbox-field/index.md (hunk 4 above)
Comment (ynotdraw) [reply docs/components/checkbox-field/index.md:316]: Nice catch - got docs all updated at https://github.com/CrowdStrike/ember-toucan-core/pull/132/commits/00cf88e3db95c947bf634a20208ef63e63c9c9bb

File: docs/components/checkbox-group-field/index.md [hunk 6]
```diff
@@ -157,6 +157,46 @@ To disable individual checkbox fields, set the `@isDisabled` argument directly o
...
+<Form::Fields::CheckboxGroup
+  @label='Label'
+  @name='options'
+  @value={{this.groupValue}}
+  @onChange={{this.updateValue}}
+  @isReadOnly={{true}}
+  as |group|
+>
+  <!-- This will now be readonly as well! -->
+  <group.CheckboxField @label='Option 1' @value='option-1' />
```
Comment (nicolechung) [inline docs/components/checkbox-group-field/index.md:176]: 🎉

File: test-app/tests/integration/components/radio-group-field-test.gts [hunk 7]
```diff
@@ -119,6 +119,55 @@ module('Integration | Component | Fields | RadioGroup', function (hooks) {
...
+
+    assert.dom('[data-radio-1]').hasAttribute('readonly');
+  });
+
+  test('it sets readonly on the fieldset and all child radios using `@isReadOnly` at the root', async function (assert) {
+    await render(<template>
+      <RadioGroupField
+        @label="Label"
+        @name="group"
+        @isReadOnly={{true}}
```
Comment (nicolechung) [inline test-app/tests/integration/components/radio-group-field-test.gts:157]: Thanks for adding this test! Seems like the DIL link is behind (at least when I'm checking it now) so it's nice to see tests that confirm this!

**Pull request merged**